### PR TITLE
python: added iio_readdev.py and iio_writedev.py

### PR DESCRIPTION
**Pull request by cristi-iacob**
Implemented the iio_readdev and iio_writedev modules with Python bindings. These modules can be used as examples for sending and receiving buffers.

### CODE REVIEW

**Comment by rgetz** [comment]
Since this will be an example - can we add some sphinx comments, so when we merge #393 things show up?

File: bindings/python/examples/iio_writedev.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+
+signal.signal(signal.SIGINT, keyboard_interrupt_handler)
+
+
+def main():
+    raw_arguments = sys.argv[1:]
+
+    try:
+        arguments, values = getopt.getopt(raw_arguments, unixOptions, gnuOptions)
```
Comment (tfcollins) [inline bindings/python/examples/iio_writedev.py:99]: Why was argparse not used? It much much simpler and in the core library.

File: bindings/python/examples/iio_writedev.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    num_samples_set = num_samples > 0
+
+    while app_running:
+        bytes_to_read = buffer._length if not num_samples_set else min(buffer._length, num_samples * dev.sample_size)
+        write_len = bytes_to_read
+        data = []
+
+        while bytes_to_read > 0:
+            data = sys.stdin.buffer.read(bytes_to_read)
+            bytes_to_read -= len(data)
```
Comment (tfcollins) [inline bindings/python/examples/iio_writedev.py:197]: Won't this loop simply overwrite data when bytes_to_read is not 0 after the first loop? Therefore the first array is never passed to the write buffer.

File: bindings/python/examples/iio_writedev.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+            if num_samples == 0 and write_len < buffer_size * dev.sample_size:
+                exit(0)
+
+        ret = buffer.write(bytearray(data))
+        buffer.push()
+
+        if ret == 0:
+            if app_running:
+                print('Unable to push buffer!')
```
Comment (tfcollins) [inline bindings/python/examples/iio_writedev.py:213]: Should this be an exception?

File: bindings/python/examples/iio_readdev.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        for channel in dev.channels:
+            channel.enabled = True
+    else:
+        for channel_idx in channels:
+            dev.channels[int(channel_idx)].enabled = True
+
+    buffer = iio.Buffer(dev, buffer_size)
+
+    if buffer is None:
+        print('Unable to create buffer!')
```
Comment (tfcollins) [inline bindings/python/examples/iio_readdev.py:179]: Should this be an exception?

**Comment by rgetz** [comment]
Also - until we have it fixed, check out: https://app.codacy.com/manual/commodo/libiio/pullRequest?prid=5299649#newIssuesView

File: bindings/python/examples/iio_writedev.py (hunk 2 above)
Comment (cristi-iacob) [reply bindings/python/examples/iio_writedev.py]: Yes, you are right.  I have modified; I think this modification should solve the problem.

File: bindings/python/examples/iio_writedev.py (hunk 1 above)
Comment (cristi-iacob) [reply bindings/python/examples/iio_writedev.py]: I didn't know argparse was in the core library; getopt just seemed simpler at the first glance for me. I have changed them to use argparse.

**Comment by rgetz** [comment]
There still doesn't seem to be any markdown in the code that will help sphinx? Should there be?

**Comment by cristi-iacob** [comment]
I have added some documentation for the functions from iio_readdev.py and iio_writedev.py.
Also I have added the .rst files for them, in order to be compatible with sphinx. If these files shouldn't have been added, I will delete the last commit.

**Comment by tfcollins** [comment]
Can you add references to the index file to the examples? Otherwise I don't think you can get to them.

**Comment by cristi-iacob** [comment]
I have added a new component on the main page, Tools. It contains references to iio_readdev and iio_writedev documentation pages.

**Comment by rgetz** [comment]
Thanks - looks great.

**Pull request merged**